sql/server.go

```diff
@@ -139,81 +68,16 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	bytes, err := gogoproto.Marshal(&planMaker.session)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+		return args.CreateReply(), http.StatusInternalServerError, err
 	}
 	reply.Session = bytes
 
-	// Marshal the response.
-	body, contentType, err := util.MarshalResponse(r, &reply, allowedEncodings)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-	w.Header().Set(util.ContentTypeHeader, contentType)
-	w.Write(body)
-}
-
-type parameters []driver.Datum
-
-// Arg implements the Args interface
-func (p parameters) Arg(name string) (parser.Datum, bool) {
-	if len(name) == 0 {
-		// This shouldn't happen unless the parser let through an invalid parameter
-		// specification.
-		panic(fmt.Sprintf("invalid empty parameter name"))
-	}
-	if ch := name[0]; ch < '0' || ch > '9' {
-		// TODO(pmattis): Add support for named parameters (vs the numbered
-		// parameter support below).
-		return nil, false
-	}
-	i, err := strconv.ParseInt(name, 10, 0)
-	if err != nil {
-		return nil, false
-	}
-	if i < 1 || int(i) > len(p) {
-		return nil, false
-	}
-	arg := p[i-1].GetValue()
-	if arg == nil {
-		return parser.DNull, true
-	}
-	switch t := arg.(type) {
-	case *bool:
-		return parser.DBool(*t), true
-	case *int64:
-		return parser.DInt(*t), true
-	case *float64:
-		return parser.DFloat(*t), true
-	case []byte:
-		return parser.DString(t), true
-	case *string:
-		return parser.DString(*t), true
-	default:
-		panic(fmt.Sprintf("unexpected type %T", t))
-	}
-}
-
-// If we hit an error and there is a pending transaction, rollback
-// the transaction before returning. The client does not have to
-// deal with cleaning up transaction state.
-func rollbackTxnAndReturnResultWithError(planMaker *planner, err error) driver.Result {
-	if planMaker.txn != nil {
-		// What do we do with a rollback error? This is an internally
-		// initiated rollback that the client is unaware of. Reporting it
-		// will only cause confusion. Not reporting it could leave a transaction
-		// pending, but that will get GCed eventually.
-		_ = planMaker.txn.Rollback()
-	}
-	var errProto proto.Error
-	errProto.SetResponseGoError(err)
-	return driver.Result{Error: &errProto}
+	return reply, 0, nil
 }
 
 // exec executes the request. Any error encountered is returned; it is
 // the caller's responsibility to update the response.
-func (s *Server) exec(req driver.Request, planMaker *planner) driver.Response {
+func (s server) exec(req driver.Request, planMaker *planner) driver.Response {
```

Comment:
i've named this `execStmts` and left `execStmt` with the same name. I took the `req` argument out of both though, so they don't care about the wire type any more.